chat_server/index.ts

```diff
@@ -15,38 +22,45 @@ class Message {
 const app = express();
 app.set("port", process.env.PORT || 3000);
 
-let httpServer = createServer(app);
-let io = socketIo(httpServer);
+export let httpServer = createServer(app);
+export let io = require('socket.io')(httpServer);
 
-io
-.on('connection', socketioJwt.authorize({
-	secret: 'your secret or public key',
-	handshake: true,
-	decodedPropertyName: ""
-}))
-.on("authenticated", (socket: any) => {
+// Authentification using handshake
+io.use(socketio_jwt.authorize(jwt_option));
 
-	socket.on("message", (request: any) => {
+io.sockets
+.on('connection', (socket) => {
+
+	socket.on("message", async (request: any) => {
 		if (request.message === undefined ){
 			return
 		}
 
 		let model = new Message(socket.decoded_token.username, request.message, new Date());
-		add(model.user_id, model.message);
-		socket.broadcast(model);
+		await db.add(model.username, model.message);
+
+		// The message is sended to everyone (including sender)
+		io.sockets.emit('broadcast', model);
 	});
 
-	socket.on("fetch", (request: any) => {
+	socket.on("fetch", async (request: any) => {
 		if (request.from === undefined || request.limit === undefined){
-			return 
+			return
 		}
 
-		let messages = await get(request.from, request.limit);
-		socket.emit("fetch_response", messages);
+		let messages = await db.get(db.dateToMysql(request.from), request.limit);
```

Comment:
`db.dateToMysql` should be encapsulated by the `get` method since it's proper to `mysql`